CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+* If you fix a bug:
+    * Write the test the reproduces the bug.
+    * Fixes without tests are accepted only in exceptional circumstances if it can be shown that writing the
+      corresponding test is too hard or otherwise impractical.
+* If you introduce any new public APIs:
+    * All new APIs must come with documentation and tests.
+    * If you plan API additions, then please start by submitting an issue with the proposed API design
+      to gather community feedback.
+    * [Contact the maintainers](#contacting-maintainers) to coordinate any big piece of work in advance via submitting an issue.
+* If you fix documentation:
+    * If you plan extensive rewrites/additions to the docs, then please [contact the maintainers](#contacting-maintainers)
+      to coordinate the work in advance.
+
+## PR workflow
+
+0. Contributor build the library locally and run all unit test via Gradle task `dataframe:test`
+   (see the ["Building"](#building) chapter).
+1. Contributor submits the PR if the local build is successful and tests are green.
+2. Reviewer put his name to the "Reviewers" section of the proposed PR at the start of the review process.
+3. Reviewer leaves the comments or marks the PR with the word "LGTM."
+4. Contributor answers the comments or fixes the proposed PR.
+5. Reviewer marks the PR with the word "LGTM."
+6. Maintainer could suggest merging the master branch to the PR branch a few times due to changes in the `master` branch.
+7. Maintainer runs TC builds (unit tests and examples as integration tests).
+8. The TC writes the result (passed or not passed) to the PR checks at the bottom of the proposed PR.
+9. If it is possible, maintainers share the details of the failed build with the contributor.
+10. Maintainer merges the PR if all checks are successful and there is no conflict with the master branch.
+
+## How to fix an existing issue
+
+* If you are going to work on the existing issue:
+    * Comment on the existing issue if you want to work on it.
+    * Wait till it is assigned to you by [maintainers](#contacting-maintainers).
+    * Ensure that the issue not only describes a problem, but also describes a solution that had received a positive feedback. Propose a solution if there isn't any.
+* If you are going to submit your first PR in this project:
+    * Find tickets with the label ["good first issue"](https://github.com/Kotlin/dataframe/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22+no%3Aassignee)
+      which are not assigned to somebody.
+    * Learn the [`examples`](https://github.com/Kotlin/dataframe/tree/master/examples) module, submit new interesting example or improve documentation for one of them.
+* If you are ready to participate in library design and in new experiments, find tickets with the label
+  ["research"](https://github.com/Kotlin/dataframe/issues?q=is%3Aissue+is%3Aopen+label%3Aresearch)
+  or join to our [discussions](https://github.com/Kotlin/dataframe/discussions).
```

Comment:
I'm not sure about it, do people really use this functionality? Existing practise is discussions in issues.

Reply:
It depends on the community, but it's very good indexing and could be found via Google and some issues like our internal stuff could be moved to the discussions too

Reply:
Hmm, we can try to discuss some specific topic there :D I for instance found information from kotlinx.coroutines issues with Google as well. My concern is that having 2 different places where you can start a conversation as an external contributor raises the bar for indecisive people. But moving our discussions there might be a good idea